### PR TITLE
Swap out pystac recursion, lint recursively

### PR DESCRIPTION
**Pull request by jonhealy1**
This swaps out pystac to use stac-validator for recursion instead. It also adds functionality to lint recursively.

https://github.com/stac-utils/stac-check/issues/82

https://github.com/stac-utils/stac-check/issues/85

### CODE REVIEW

File: stac_check/lint.py [hunk 1]
```diff
@@ -18,15 +17,16 @@ class Linter:
...
+    recursive: int = 0
 
     def __post_init__(self):
         self.data = self.load_data(self.item)
         self.message = self.validate_file(self.item)
         self.config = self.parse_config(self.config_file)
         self.asset_type = self.message["asset_type"] if "asset_type" in self.message else ""
         self.version = self.message["version"] if "version" in self.message else ""
-        self.validator_version = "2.3.0"
+        self.validator_version = "3.0.0"
```
Comment (gadomski) [inline stac_check/lint.py:28]: Can we get this from `stac-validator`, e.g. via `stac_validator.__version__` or this more refined version:

```python
import pkg_resources
self.validator_version = pkg_resources.require("stac_validator")[0].version
```

File: stac_check/lint.py [hunk 2]
```diff
@@ -18,15 +17,16 @@ class Linter:
     config_file: Optional[str] = None
     assets: bool = False
     links: bool = False
-    recursive: bool = False
+    recursive: int = 0
```
Comment (gadomski) [inline stac_check/lint.py:20]: In `stac-validator` we separated the recursion flag from max depth, e.g.: https://github.com/stac-utils/stac-validator/blob/f8647f8ce3ac44e355d68bdf3f17cfc9914f689c/stac_validator/stac_validator.py#L31-L42. IMO we should use the same pattern here?

**Review by gadomski — CHANGES_REQUESTED** [review]

File: stac_check/lint.py (hunk 2 above)
Comment (jonhealy1) [reply stac_check/lint.py:20]: true

File: stac_check/lint.py (hunk 1 above)
Comment (jonhealy1) [reply stac_check/lint.py]: great tip thanks!

File: stac_check/cli.py [hunk 3]
```diff
@@ -61,7 +78,7 @@ def cli_message(linter):
     if linter.validate_all == True:
         click.secho()
         click.secho(f"Recursive validation has passed!", fg='blue')
-    elif linter.validate_all == False and linter.recursive == True:
+    elif linter.validate_all == False and linter.recursive > 0:
```
Comment (gadomski) [inline stac_check/cli.py:81]:
```suggestion
    elif linter.validate_all == False and linter.recursive:
```

**Review by gadomski — CHANGES_REQUESTED** [review]

**Pull request merged**